server/util/metricsutil/metricsutil.go

```diff
@@ -0,0 +1,23 @@
+package metricsutil
+
+import (
+	"slices"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/flag"
+)
+
+var (
+	observedGroupIDs = flag.Slice("metrics.observed_group_ids", []string{}, "Group IDs allowed in group_id metrics labels. Groups not in this list will be labeled as 'ANON' if unauthenticated or 'other' if authenticated.")
+)
+
+// GroupIDLabel returns a value suitable for use with the GroupID label. It
+// restricts the observed group IDs to the allowed values configured by flag.
+func GroupIDLabel(groupID string) string {
```

Comment:
Can we call this FilteredGroupIDLabel? There will still be cases where we will want unfiltered group IDs on metrics.

Reply:
Done